### ISSUE TITLE
Eliminate cross-repo `docker-osxcross-rust` dependency from build-scripts

### ISSUE THREAD

**Issue by kantai** [stacks-network/stacks-blockchain#3208]
The build scripts currently use a release output from https://github.com/hirosystems/docker-osxcross-rust -- this works okay, but ultimately is pretty brittle if this repo ever wanted to update that dependency.

It would be better if that dependency at least had a build script in this repo (e.g., in `./build-scripts/osxcross-rust/..`) and published the tarball in stacks-network/stacks-blockchain rather than through the separate repo. Another potential option is to include that build in the repo's build pipeline, but because we don't expect that dependency to change often, it may be very wasteful.

**Comment by kantai** [comment]
@wileyj

**Comment by wileyj** [comment]
I agree, i had a similar issue trying to build for arm64 with musl... had
to use an external dependency that could likewise be brittle.
definitely will take a look at this

**Comment by wileyj** [comment]
#4168
